client/src/search/Typeahead/Typeahead.js

```diff
@@ -40,60 +53,69 @@ export class Typeahead extends React.Component {
   componentDidUpdate(prevProps, prevState) {
-    const { page_size } = this.props;
-
-    const { query_value, current_search_configs } = this.state;
+    const {
+      query_value,
+      current_search_configs,
+      selection_cursor,
+    } = this.state;
     const {
       query_value: prev_query_value,
       current_search_configs: prev_search_configs,
+      selection_cursor: prev_selection_cursor,
     } = prevState;
 
     if (
       query_value !== prev_query_value ||
       current_search_configs !== prev_search_configs
     ) {
-      const matching_results_by_page = !this.show_menu
+      const matching_results = !this.show_menu
         ? []
-        : _.chain(this.all_options)
-            .filter(({ config_group_index, data }) =>
-              // could use optional chaining, but we WANT this to fail fast and loud, to catch
-              // malformed search_configs during development. Should be safe otherwsie
-              this.config_groups[config_group_index].group_filter(
-                query_value,
-                data
-              )
+        : _.filter(this.all_options, ({ config_group_index, data }) =>
+            // could use optional chaining, but we WANT this to fail fast and loud, to catch
+            // malformed search_configs during development. Should be safe otherwsie
+            this.config_groups[config_group_index].group_filter(
+              query_value,
+              data
             )
-            .chunk(page_size)
-            .value();
+          );
 
       this.setState({
-        matching_results_by_page,
-        pagination_cursor: this.default_pagination_cursor,
+        matching_results,
         selection_cursor: this.default_selection_cursor,
       });
-    } else {
-      const active_item = this.active_item;
+    }
+
+    if (this.list_ref.current) {
+      this.list_ref.current.Grid.measureAllCells();
+      const list_height = this.list_ref.current.Grid.getTotalRowsHeight();
+      if (list_height !== this.state.list_height) {
+        if (list_height < 400) {
+          this.setState({ list_height: list_height });
+        } else if (this.state.list_height < 400) {
+          this.setState({ list_height: 400 });
+        }
+      }
 
-      if (active_item) {
-        active_item.scrollIntoView({
-          behaviour: "auto",
-          block: "nearest",
-        });
-      } else {
-        this.typeahead_ref.current
-          .querySelector(".typeahead__dropdown > *")
-          ?.scrollIntoView({
-            behaviour: "auto",
-            block: "nearest",
-          });
+      if (
+        query_value !== prev_query_value ||
+        current_search_configs !== prev_search_configs
+      ) {
+        cache.clearAll();
+      }
+
+      if (
+        selection_cursor !== prev_selection_cursor &&
+        prev_selection_cursor > selection_cursor
+      ) {
+        this.list_ref.current.recomputeRowHeights(selection_cursor); //scrolling up is choppy if we don't do this
```

Comment:
Can you quickly explain what's happening from line 94 to here?

Reply:
Yea, so if we treat this as 3 separate sections, we have a height manager, and 2 cell-height refreshers.

### Height manager
For this section (lines 97 - 105), we are checking to see what the height of the list should be. A hard value of height has to be passed into `List` otherwise it will render with height 0. Thus we check if the height of all cells combined will be greater than 400px, and if not, we have to change the height we pass into `List`. The value 400 is taken from the typeahead max height in master.

###cache.clearAll()
Height for each cell of `List` is cached to prevent unnecessary recalculations (assuming height doesn't change). This was implemented since changing search configs were messing with the heights.

### Recompute height (refer to issue 610 of react-virutalized)
There's a known issue that causes jumping when scrolling from the bottom to the top. This was added so that on pressing the up arrow, it would fix the stuttering. Oddly enough, regular scrolling doesn't trigger the jumping issue for us though?

We'll want to remove that last bit if it ever gets fixed

Hope that clarifies it!